### ISSUE TITLE
[GSSoC 25] Suggestion: Integrate Hyperparameter Tuning for LSTM Model using Optuna

### ISSUE THREAD

**Issue by ManjuBodi** [sahithinandikula/Sneaker-price-prediction#44]
<p><strong>Context:</strong></p>
<p>While reviewing the current LSTM-based sneaker price prediction implementation, I noticed that the model is using fixed/default hyperparameters (e.g., hidden size, learning rate, number of layers, etc.). These can have a significant impact on performance and generalization.</p>

<p>Introducing a hyperparameter tuning framework like <strong>Optuna</strong> could lead to substantial improvements in model accuracy, convergence speed, and overall robustness.</p>

<hr>

<h3> Why Use Optuna?</h3>
<ul>
  <li><strong>Efficient Search:</strong> Uses smarter sampling (Tree-structured Parzen Estimators) to explore the hyperparameter space efficiently.</li>
  <li><strong>Pruning:</strong> Can terminate unpromising trials early to save compute time.</li>
  <li><strong>Lightweight Integration:</strong> Easy to add on top of existing PyTorch training loops.</li>
  <li><strong>Visualization:</strong> Offers built-in tools to visualize optimization history, parameter importance, and more.</li>
</ul>

<hr>

<h3> Expected Outcome:</h3>
<ul>
  <li>Improved predictive performance (lower MAE/RMSE/MAPE)</li>
  <li>More stable training and potentially faster convergence</li>
  <li>Better understanding of model sensitivity to hyperparameters</li>
</ul>

<hr>

<h3> Request:</h3>
<p>If this sounds good, I’d like to volunteer to implement the Optuna-based tuning framework. Please feel free to assign this issue to me.</p>

<p>Happy to collaborate or align with any constraints (compute time, budget, etc.) before starting.</p>

**Comment by Sathvika1507** [comment]
Hi, I’m Sathvika (GSSoC’25 contributor). I’d like to work on this issue as my first contribution. Could you please assign it to me?
